src/Formik.tsx

```diff
@@ -411,10 +411,14 @@ export class Formik<Values = {}, ExtraProps = {}> extends React.Component<
       }
     }
 
-    this.submitForm();
+    this.submitForm().catch(error => {
+      if (error.name !== 'ValidationError') {
+        throw error;
```

Comment:
Why do we want to throw this? Why not just set it to `apiError` or something?

Reply:
We could do that, but the error does not necessarily have to be from an api. It could be a network error, for example. Also, an api might be responding with specific http status codes that the user might need to just catch and handle accordingly. I do not see the benefit of hiding the exception.

Reply:
https://en.m.wikipedia.org/wiki/Error_hiding

Reply:
How would the user catch this? People who do SSR cannot use ErrorBoundary / componentDidCatch?

Reply:
You can see an example in #486.

Reply:
@rhyek according to my understanding, ValidationError is something related to "yup" only, and this.submitForm() does not throw validations "combinedErrors" when there are errors (in yup or not), so this catch should basically do nothing. Note: runValidations currently does not reject on errors

I suggest making "submitForm" throw" on any error (validation + async calls + others), and catching here all "expected exceptions" (ie validation/network errors) and rethrow unexpected ones so that global error handler can log them (ie report to sentry or something). Maybe this strategy can be pluggable so that user may customize it to his own needs.

Reply:
I am not sure, to be honest. Looking at the code quickly it seems there's some new things I hadn't seen before. This is PR is kind of old.